crates/ruff/src/rules/pep8_naming/rules/invalid_function_name.rs

```diff
@@ -0,0 +1,76 @@
+use ruff_macros::{define_violation, derive_message_formats};
+use rustpython_parser::ast::Stmt;
+
+use crate::ast::helpers::identifier_range;
+use crate::registry::Diagnostic;
+use crate::source_code::Locator;
+use crate::violation::Violation;
+
+define_violation!(
+    /// ## What it does
+    ///
+    /// Checks for functions names that are not lowercase, with words separated by underscores.
```

Comment:
This convention is commonly known as snake_case.

Reply:
Updated all to consistently use CamelCase and snake_case. To remove any remaining ambiguity, there is a wiki link included.